### DISCUSSION TITLE
Multiple clusters with same name in VMWare

### DISCUSSION THREAD

**Discussion by deltabrav0** [bb-Ricardo/netbox-sync#230] [Q&A]
I think this [issue](https://github.com/bb-Ricardo/netbox-sync/issues/119) is closed, but I'm having the same issue running the development release.

I have multiple clusters with the same name:

- TX01
    - CL01
    - CL02
    - CL03
- IL01
    - CL01

I have two vmware sources: TX01 and IL01, TX01 is the first, IL01 is the second
Each source has the following stanza
cluster_site_relation = TX01/* = LFKNTXXA, IL01/* = MTONILXC

After discovery, there is no CL01 cluster in TX01 group. I believe all the CL01 clusters get overwritten from the discovery of IL01 since it is the second source definition.

Am I missing something?

**Comment by lukify85** [comment]
I'm having a similar issue. I have a "Cluster A" in both datacenters. I've tried to define the datacenter in the "cluster_site_relation" field in several ways and the issue remains.

cluster_site_relation = "FLOR Horizon/.*" = FLOR_SITE
cluster_site_relation = FLOR Horizon/.* = FLOR_SITE
cluster_site_relation = FLOR Horizon/* = FLOR_SITE
cluster_site_relation = "FLOR Horizon"/* = FLOR_SITE
cluster_site_relation = "FLOR Horizon"/"Cluster A" = FLOR_SITE
cluster_site_relation = FLOR Horizon/Cluster A = FLOR_SITE

Same or similar result every time. FLTA overwrites since it is the subsequent VCSA instance being scannned.

~~~~~~~~~~~~~~~

The following an abridged verson of my settings.ini

; This option defines which vCenter cluster is part of a NetBox site.
; This is done with a comma separated key = value list.
;   key: defines the cluster name as regex
;   value: defines the NetBox site name (use quotes if name contains commas)
; This is a quite important config setting as IP addresses, prefixes, VLANs
; and VRFs are site dependent. In order to assign the correct prefix to an IP
; address it is important to pick the correct site.
; A VM always depends on the cluster site relation
; a cluster can be specified as "Cluster-name" or
; "Datacenter-name/Cluster-name" if multiple clusters have the same name.
; When a vCenter cluster consists of hosts from multiple NetBox sites,
; it is possible to leave the site for a NetBox cluster empty. All VMs from
; this cluster will then also have no site reference.
; The keyword "<NONE>" can be used as a value for this.
;cluster_site_relation = Cluster_NYC = New York, Cluster_FFM.* = Frankfurt, Datacenter_TOKIO/.* = Tokio, Cluster_MultiSite = <NONE>

[source/FLOR_VDIvCenter]

enabled = True
type = vmware
host_fqdn = FLOR-*******.*****.com
username = ******@*****.com
password = *****
validate_tls_certs = False
cluster_site_relation = "FLOR Horizon/.*" = FLOR_SITE
vm_role_relation = .* = VDI
dns_name_lookup = True
skip_vm_templates = True
skip_offline_vms = False
strip_host_domain_name = False
strip_vm_domain_name = False
permitted_subnets = !172.16.0.0/12, 10.0.0.0/8, !192.168.0.0/16, !fd00::/8, !10.23.42.0/24

[source/FLTA_VDIvCenter]

enabled = True
type = vmware
host_fqdn = FLTA-*******.*****.com
username = *****@*****.com
password = *****
validate_tls_certs = False
cluster_site_relation = "FLTA Horizon/.*" = FLTA_SITE
vm_role_relation = .* = VDI
dns_name_lookup = True
skip_vm_templates = True
skip_offline_vms = False
strip_host_domain_name = False
strip_vm_domain_name = False
permitted_subnets = !172.16.0.0/12, 10.0.0.0/8, !192.168.0.0/16, !fd00::/8, !10.23.42.0/24

**Reply by bb-Ricardo** [reply]
Hey,

Just wondering if you have created the clusters and site attached in Netbox before the sync. That could be the reason.

Would also write it this way:
`cluster_site_relation = "FLOR Horizon/Cluster A" = FLOR_SITE`

**Comment by lukify85** [comment]
That was indeed the reason. Thank you! Wonderful tool you've built! And thank you for the clarification on the cluster_site_relation syntax.

**Reply by bb-Ricardo** [reply]
Great, glad you could figure it out.
